aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -165,18 +167,58 @@ class CAFFE2_API SchemaRegistrationHandleRAII final {
 template<class Return, class... Args>
 inline Return Dispatcher::callUnboxed(const OperatorHandle& op, Args... args) const {
   // note: this doesn't need the mutex because write operations on the list keep iterators intact.
-  return op.operatorIterator_->op.callUnboxed<Return, Args...>(std::forward<Args>(args)...);
+  return op.operatorIterator_->op.readDispatchTable([&] (const DispatchTable& dispatchTable) -> Return {
+    c10::optional<TensorTypeId> dispatchKey = dispatchTable.dispatchKeyExtractor().getDispatchKeyUnboxed(args...);
+    const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+    return kernel.template callUnboxed<Return, Args...>(std::forward<Args>(args)...);
+  });
 }
 
 template<class Return, class... Args>
 inline Return Dispatcher::callUnboxedOnly(const OperatorHandle& op, Args... args) const {
   // note: this doesn't need the mutex because write operations on the list keep iterators intact.
-  return op.operatorIterator_->op.callUnboxedOnly<Return, Args...>(std::forward<Args>(args)...);
+  return op.operatorIterator_->op.readDispatchTable([&] (const DispatchTable& dispatchTable) -> Return {
+    c10::optional<TensorTypeId> dispatchKey = dispatchTable.dispatchKeyExtractor().getDispatchKeyUnboxed(args...);
+    const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+    return kernel.template callUnboxedOnly<Return, Args...>(std::forward<Args>(args)...);
+  });
 }
 
 inline void Dispatcher::callBoxed(const OperatorHandle& op, Stack* stack) const {
   // note: this doesn't need the mutex because write operations on the list keep iterators intact.
-  return op.operatorIterator_->op.callBoxed(stack);
+  return op.operatorIterator_->op.readDispatchTable([&] (const DispatchTable& dispatchTable) {
+    c10::optional<TensorTypeId> dispatchKey = dispatchTable.dispatchKeyExtractor().getDispatchKeyBoxed(stack);
+    const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+    kernel.callBoxed(stack);
+  });
+}
+
+inline const KernelFunction& Dispatcher::dispatch_(const DispatchTable& dispatchTable, c10::optional<TensorTypeId> dispatchKey) const {
```

Comment:
this was already inline before, just in a different file. I did a quick check on libtorch.so to confirm and actually saw a small binary size improvement from this PR, potentially because I shortened some of the call chains.